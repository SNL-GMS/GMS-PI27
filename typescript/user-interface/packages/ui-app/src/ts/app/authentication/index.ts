import { ianAuthenticator } from './authentication';
import type { Authenticator } from './types';

export const authenticator: Authenticator = ianAuthenticator;
