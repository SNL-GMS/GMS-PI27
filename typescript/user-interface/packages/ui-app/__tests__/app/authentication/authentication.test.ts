import { ianAuthenticator } from '../../../src/ts/app/authentication/authentication';

describe('ian-authentication', () => {
  test('imports should be defined', () => {
    expect(ianAuthenticator).toBeDefined();
  });

  test('ianAuthenticator should have defined object fields', () => {
    /* eslint-disable @typescript-eslint/unbound-method */
    expect(ianAuthenticator.authenticateWith).toBeDefined();
    expect(ianAuthenticator.checkIsAuthenticated).toBeDefined();
    expect(ianAuthenticator.unAuthenticateWith).toBeDefined();
    expect(ianAuthenticator.logout).toBeDefined();
    /* eslint-enable @typescript-eslint/unbound-method */
  });

  test("authenticateWith('test') should return an authenticated session", async () => {
    const result = await ianAuthenticator.authenticateWith('test');
    expect(result.authenticated).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  test('authenticateWith() should return an authenticated session', async () => {
    const result = await ianAuthenticator.authenticateWith('username');
    expect(result.authenticated).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  test('authenticateWith() should return an unauthenticated session', async () => {
    const result = await ianAuthenticator.authenticateWith('');
    expect(result.authenticated).toBeFalsy();
    expect(result).toMatchSnapshot();
  });

  test('checkIsAuthenticated should return an authenticated session', async () => {
    const result = await ianAuthenticator.checkIsAuthenticated();
    expect(result.authenticated).toBeTruthy();
    expect(result).toMatchSnapshot();
  });

  test('logout should call back with an unauthenticated session', () => {
    ianAuthenticator.logout(result => {
      expect(result.authenticated).toBeFalsy();
      expect(result).toMatchSnapshot();
    });
  });

  test('checkIsAuthenticated should return an unauthenticated session', async () => {
    const result = await ianAuthenticator.checkIsAuthenticated();
    expect(result.authenticated).toBeFalsy();
    expect(result).toMatchSnapshot();
  });

  test('unauthenticated if logout should fail', () => {
    ianAuthenticator.logout(result => {
      // eslint-disable-next-line no-eval
      eval('ðŸ’£'); // Exercise the "catch" of the logout
      expect(result.authenticated).toBeFalsy();
    });
  });

  test('logs an error if logout should fail', () => {
    const errorMessage = 'Failed to un - authenticate: SyntaxError: Invalid or unexpected token';
    ianAuthenticator.logout(() => {
      // eslint-disable-next-line no-eval
      eval('ðŸ’£'); // Exercise the "catch" of the logout
      console.error = jest.fn();
      expect(console.error).toHaveBeenCalledWith(errorMessage);
    });
  });
});
