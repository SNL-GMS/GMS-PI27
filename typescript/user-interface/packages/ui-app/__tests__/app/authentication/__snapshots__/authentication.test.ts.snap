// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ian-authentication authenticateWith('test') should return an authenticated session 1`] = `
{
  "authenticated": true,
  "authenticationCheckComplete": true,
  "failedToConnect": false,
  "userName": "test",
}
`;

exports[`ian-authentication authenticateWith() should return an authenticated session 1`] = `
{
  "authenticated": true,
  "authenticationCheckComplete": true,
  "failedToConnect": false,
  "userName": "username",
}
`;

exports[`ian-authentication authenticateWith() should return an unauthenticated session 1`] = `
{
  "authenticated": false,
  "authenticationCheckComplete": true,
  "failedToConnect": false,
  "userName": "",
}
`;

exports[`ian-authentication checkIsAuthenticated should return an authenticated session 1`] = `
{
  "authenticated": true,
  "authenticationCheckComplete": true,
  "failedToConnect": false,
  "userName": "username",
}
`;

exports[`ian-authentication checkIsAuthenticated should return an unauthenticated session 1`] = `
{
  "authenticated": false,
  "authenticationCheckComplete": true,
  "failedToConnect": false,
  "userName": "",
}
`;

exports[`ian-authentication logout should call back with an unauthenticated session 1`] = `
{
  "authenticated": false,
  "authenticationCheckComplete": true,
  "failedToConnect": false,
  "userName": "default",
}
`;
